src/App.tsx

```diff
@@ -1,7 +1,33 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  key: string;
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (e) => {
+      this.setState({ key: e.key });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.setState, false);
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <p className="App__message">
+          {this.state.key
```

Comment:
you could restructure this.state.key to make it look nicer

Reply:
Yeah, just make a destructuring from this.state